src/Engine/Component/Script.cpp

```diff
@@ -45,35 +45,41 @@ Json::Value Script::Save() const {
     return component;
 }
 
-void Script::FillPropertyMap() {
+void Script::AddToPropertyMap(std::string &name, int type, int size, void* data) {
```

Comment:
Ampersand of `name` should be adjacent to `std::string`. That parameter could also be made const (because we are not mutating the reference provided to us).